charts/kube-state-metrics/templates/role.yaml

```diff
@@ -1,180 +1,192 @@
-{{- if .Values.rbac.create -}}
+{{- if and (eq $.Values.rbac.create true) (not .Values.rbac.useExistingRole)  -}}
+{{- if eq .Values.rbac.useClusterRole false }}
+{{-   range (split "," $.Values.namespace) }}
+{{- end }}
+---
 apiVersion: rbac.authorization.k8s.io/v1
+{{- if eq .Values.rbac.useClusterRole false }}
```

Comment:
Can we switch branches here to avoid checking for false?

Reply:
How do you mean "switch branches" ?
I check for the same "exact thing" ( if eq .Values.rbac.useClusterRole false ) - in the entire file - to be certain it either "loops over namespaces" or goes "cluster-wide" way.
Do you mean I test for positive instead..  ie.( if .Values.rbac.useClusterrole )  .
Why do you dislike testing specificly for false? as I understand it, it follows these rules: https://godoc.org/text/template - ie. "false, 0, any nil pointer or interface value, and any array, slice, map, or string of length zero." is considered to match false.

IMHO doing it this way, removes ambiguity.

Reply:
Yes my idea was to check for true instead of false but you convinced me. Thanks for your clarification.